### DISCUSSION TITLE
Zarr's Supported Indexing

### DISCUSSION THREAD

**Discussion by ilan-gold** [zarr-developers/zarr-python#2485] [General]
Hello,

I am not entirely clear about what is supported and what isn't in zarr indexing, especially in v3. I understand https://zarr.readthedocs.io/en/stable/tutorial.html#advanced-indexing as "the API is different because we explicitly implement oindex but otherwise the behavior should be the same as numpy."

In other words, it seems "coordinate indexing" -> `vindex` (or just normal bracket shortcutting on the array directly) and "orthogonal indexing" -> `oindex`.

But other than these things, I'm not sure I understand what is and isn't supported.  For example

```python
import tempfile
import numpy as np
import zarr
from zarr.storage import LocalStore

shape = (4, 4, 4)
chunks = (2, 2, 2)
fill_value = 32767

tmp = tempfile.TemporaryDirectory()
arr = zarr.create(
    shape,
    store=LocalStore(root=tmp.name, mode="w"),
    chunks=chunks,
    dtype=np.int16,
    fill_value=fill_value,
    codecs=[zarr.codecs.BytesCodec(), zarr.codecs.BloscCodec()],
)
index_tuple = (np.array([1, 2]), np.array([1, 2]), slice(None))
# works
np.arange(64).reshape(shape)[index_tuple]
# breaks
arr[index_tuple]
```
This is on zarr v3 `3.0.0b1`.  Is this a bug?  Is there any good documentation on this?

**Comment by d-v-b** [comment]
A lot of the indexing code in 3.x is derived from indexing code in 2.x, so if something worked in zarr-python 2 it _should_ work in 3. I would guess that what you are seeing is very likely a bug, and I don't think we have good docs for indexing in 3.x, sorry about that!

**Reply by ilan-gold** [reply]
Just confirmed the bug exists in zarr 2.18.3 as well:

```python
import zarr
import numpy as np

index_tuple = (np.array([1, 2]), np.array([1, 2]), slice(None))
arr = zarr.open(shape=(4, 4, 4))
arr[index_tuple]
```
